rust/bindgen.bzl

```diff
@@ -0,0 +1,156 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _rust_bindgen(ctx):
+    rust_toolchain = ctx.toolchains["@io_bazel_rules_rust//rust:toolchain"]
+
+    bindgen = ctx.executable.bindgen
+    rustfmt = rust_toolchain.rustfmt if hasattr(rust_toolchain, 'rustfmt') else None
```

Comment:
I don't know if rustfmt should be expected to be present in the toolchain.

I think this should be initially set up to expect the rustfmt binary to be passed as an attr. If we want to bake this into the toolchain, we should take the toolchain as an attr.

Reply:
This may have made more sense when the toolchain was derived from the 'rust' component.

Providing the toolchain as an attr seems awkward since this needs the exec_platform toolchain, which afaiu should be looked up like this; I think it makes sense to still have rust's binary tooling associated with rust_toolchain, though I am not attached to it.

Reply:
(Just noticed this diff is missing the addition of the attribute to toolchain.bzl ...)